config/http_config.go

```diff
@@ -319,6 +319,9 @@ type HTTPClientConfig struct {
 	// The omitempty flag is not set, because it would be hidden from the
 	// marshalled configuration when set to false.
 	EnableHTTP2 bool `yaml:"enable_http2" json:"enable_http2"`
+	// HTTPHost optionally overrides the Host header to send.
+	// If empty, the host from the URL is used.
```

Comment:
How does this reconcile with the code that says:
```
rt = NewHostRoundTripper(opts.host, rt)
```

Reply:
> If empty, the host from the URL is used.

This is go default behavior. It's similar the the TLSServerName property.

If the property is not set by caller, then the value will be used from URL.

Ref: https://github.com/golang/go/blob/959b3fd4265d7e4efb18af454cd18799ed70b8fe/src/net/http/request.go#L240-L243

Reply:
Would it not be more correct to mention `opts.host` ?